superset-frontend/src/explore/components/ControlHeader.tsx

```diff
@@ -55,6 +59,18 @@ const ControlHeader: FC<ControlHeaderProps> = ({
   danger,
 }) => {
   const { gridUnit, colors } = useTheme();
+  const hasHadNoErrors = useRef(false);
+  const labelColor = useMemo(() => {
+    if (!validationErrors.length) {
+      hasHadNoErrors.current = true;
+    }
+
+    return hasHadNoErrors.current
```

Comment:
We tend to avoid doing this. It would be better to unpack this logic for better readability